src/test/groovy/edu/ie3/datamodel/io/sink/SqlSinkTest.groovy

```diff
@@ -0,0 +1,361 @@
+
+  @Shared
+  DatabaseNamingStrategy namingStrategy
+
+  @Shared
+  DbGridMetadata identifier
+
+  static String schemaName = "public"
+
+  def setupSpec() {
+    // Copy sql import scripts into docker
+    MountableFile sqlImportFile = getMountableFile("_sql/")
+    postgreSQLContainer.copyFileToContainer(sqlImportFile, "/home/")
+    postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + "setup.sql")
+
+    connector = new SqlConnector(postgreSQLContainer.jdbcUrl, postgreSQLContainer.username, postgreSQLContainer.password)
+
+    namingStrategy = new DatabaseNamingStrategy()
+
+    identifier = new DbGridMetadata("vn_simona", UUID.fromString("8e6bd444-4580-11ee-be56-0242ac120002"))
+
+    sqlSource = new SqlDataSource(connector, schemaName, namingStrategy)
+  }
+
+  def setup() {
+    // Execute import script
+    Iterable<String> importFiles = Arrays.asList(
+    "grids.sql",
+    "types.sql",
+    "result_entities.sql",
+    "input_entities.sql",
+    "time_series.sql",
+    "load_profile.sql"
+    )
+    for (String file: importFiles) {
+      Container.ExecResult res = postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + file)
+      assert res.stderr.empty
+    }
+  }
+
+  def cleanup() {
+    postgreSQLContainer.execInContainer("psql", "-Utest", "-f/home/" + "cleanup.sql")
+  }
+
+  def "SQL sink can persist provided elements correctly"() {
+    given:
+    TimeSeriesProcessor<IndividualTimeSeries, TimeBasedValue, EnergyPriceValue> timeSeriesProcessor = new TimeSeriesProcessor<>(IndividualTimeSeries, TimeBasedValue, EnergyPriceValue)
+    TimeSeriesProcessorKey timeSeriesProcessorKey = new TimeSeriesProcessorKey(IndividualTimeSeries, TimeBasedValue, EnergyPriceValue)
+    HashMap<TimeSeriesProcessorKey, TimeSeriesProcessor> timeSeriesProcessorMap = new HashMap<>()
+    timeSeriesProcessorMap.put(timeSeriesProcessorKey, timeSeriesProcessor)
+
+    SqlSink sink = new SqlSink(schemaName,
+    new ProcessorProvider([
+      new ResultEntityProcessor(PvResult),
+      new ResultEntityProcessor(WecResult),
+      new ResultEntityProcessor(EvResult),
+      new ResultEntityProcessor(EvcsResult),
+      new ResultEntityProcessor(EmResult),
+      new ResultEntityProcessor(FlexOptionsResult),
+      new InputEntityProcessor(Transformer2WInput),
+      new InputEntityProcessor(NodeInput),
+      new InputEntityProcessor(EvcsInput),
+      new InputEntityProcessor(Transformer2WTypeInput),
+      new InputEntityProcessor(LineGraphicInput),
+      new InputEntityProcessor(NodeGraphicInput),
+      new InputEntityProcessor(CylindricalStorageInput),
+      new InputEntityProcessor(ThermalHouseInput),
+      new InputEntityProcessor(OperatorInput),
+      new InputEntityProcessor(LineInput),
+      new InputEntityProcessor(ThermalBusInput),
+      new InputEntityProcessor(LineTypeInput),
+      new InputEntityProcessor(LoadInput),
+      new InputEntityProcessor(EmInput)
+    ], timeSeriesProcessorMap),
```

Comment:
Can't you use the shorthand constructor `ProcessorProvider()` or are any classes still unhandled?

Reply:
ProcessorProvider() works!